### PR TITLE
update solc npm version to 0.8.26

### LINKED ISSUE #5080: Update solc version to latest in remix-solidity
See: https://github.com/ethereum/remix-project/pull/3718

### PR DESCRIPTION
**Pull request by Aniket-Engg**
fixes #5080

### CODE REVIEW

File: libs/remix-debug/test/decoder/contracts/byteStorage.ts [hunk 1]
```diff
@@ -3,13 +3,13 @@
...
...53, e54, e55, e56, e57, e58, e59, e60, e61, e62, e63, e64, e65, e66, e67, e68, e69, e70, e71, e72, e73, e74, e75, e76, e77, e78, e79, e80, e81, e82, e83, e84, e85, e86, e87, e88, e89, e90, e91, e92, e93, e94, e95, e96, e97, e98, e99, e100, e101, e102, e103, e104, e105, e106, e107, e108, e109, e110, e111, e112, e113, e114, e115, e116, e117, e118, e119, e120, e121, e122, e123, e124, e125, e126, e127, e128, e129, e130, e131, e132, e133, e134, e135, e136, e137, e138, e139, e140, e141, e142, e143, e144, e145, e146, e147, e148, e149, e150, e151, e152, e153, e154, e155, e156, e157, e158, e159, e160, e161, e162, e163, e164, e165, e166, e167, e168, e169, e170, e171, e172, e173, e174, e175, e176, e177, e178, e179, e180, e181, e182, e183, e184, e185, e186, e187, e188, e189, e190, e191, e192, e193, e194, e195, e196, e197, e198, e199, e200, e201, e202, e203, e204, e205, e206, e207, e208, e209, e210, e211, e212, e213, e214, e215, e216, e217, e218, e219, e220, e221, e222, e223, e224, e225, e226, e227, e228, e229, e230, e231, e232, e233, e234, e235, e236, e237, e238, e239, e240, e241, e242, e243, e244, e245, e246, e247, e248, e249, e250, e251, e252, e253, e254, e255, e256, e257, e258, e259, e260 }
```
Comment (yann300) [inline libs/remix-debug/test/decoder/contracts/byteStorage.ts:6]: why did `e256, e257, e258, e259, e260` get removed?

File: libs/remix-debug/test/decoder/contracts/miscContracts.ts [hunk 2]
```diff
@@ -27,7 +27,7 @@ module.exports = `
...
...1,item152,item153,item154,item155,item156,item157,item158,item159,item160,item161,item162,item163,item164,item165,item166,item167,item168,item169,item170,item171,item172,item173,item174,item175,item176,item177,item178,item179,item180,item181,item182,item183,item184,item185,item186,item187,item188,item189,item190,item191,item192,item193,item194,item195,item196,item197,item198,item199,item200,item201,item202,item203,item204,item205,item206,item207,item208,item209,item210,item211,item212,item213,item214,item215,item216,item217,item218,item219,item220,item221,item222,item223,item224,item225,item226,item227,item228,item229,item230,item231,item232,item233,item234,item235,item236,item237,item238,item239,item240,item241,item242,item243,item244,item245,item246,item247,item248,item249,item250,item251,item252,item253,item254,item255,item256,item257,item258,item259,item260,item261,item262,item263,item264,item265,item266,item267,item268,item269,item270,item271,item272,item273,item274,item275,item276,item277,item278,item279,item280,item281,item282,item283,item284,item285,item286,item287,item288,item289,item290,item291,item292,item293,item294,item295,item296,item297,item298,item299,item100000000}
```
Comment (yann300) [inline libs/remix-debug/test/decoder/contracts/miscContracts.ts:30]: same here for  `item264,item265,item266,item267,item268,item269,item270,item271,item272,item273,item274,item275,item276,item277,item278,item279,item280,item281,item282,item283,item284,item285,item286,item287,item288,item289,item290,item291,item292,item293,item294,item295,item296,item297,item298,item299,item100000000`

File: libs/remix-debug/test/decoder/decodeInfo.ts [hunk 3]
```diff
@@ -47,7 +47,7 @@ tape('solidity', function (t) {
     state = astHelper.extractStateDefinitions('test.sol:contractEnum', output.sources, null)
     stateDef = state.stateDefinitions
     parsedType = decodeInfo.parseType(stateDef[1].typeDescriptions.typeString, states, 'contractEnum', null)
-    checkDecodeInfo(st, parsedType, 1, 2, 'enum')
```
Comment (yann300) [inline libs/remix-debug/test/decoder/decodeInfo.ts:50]: why did the storageBytes updated here?

File: libs/remix-debug/test/decoder/localsTests/structArray.ts [hunk 4]
```diff
@@ -103,7 +103,7 @@ module.exports = function (st, privateKey, contractBytecode, compilationResult,c
                 st.equals(locals['arrayStruct'].value.b.value[0].value, '34')
                 st.equals(locals['arrayStruct'].value.b.value[1].value, '-23')
                 st.equals(locals['arrayStruct'].value.b.value[2].value, '-3')
-                st.equals(locals['arrayStruct'].value.c.value, 'three')
```
Comment (yann300) [inline libs/remix-debug/test/decoder/localsTests/structArray.ts:106]: weren't you able to find an index were this var is set to `three`?

File: libs/remix-debug/test/decoder/contracts/byteStorage.ts (hunk 1 above)
Comment (Aniket-Engg) [reply libs/remix-debug/test/decoder/contracts/byteStorage.ts:6]: enum can take only 256 elements as per the new versions. See: https://docs.soliditylang.org/en/v0.8.26/080-breaking-changes.html#new-restrictions

File: libs/remix-tests/tests/examples_0/assert_greaterThan_test.sol [hunk 5]
```diff
@@ -29,8 +29,4 @@ contract AssertGreaterThanTest {
     function greaterThanIntUintPassTest() public {
         Assert.greaterThan(int(10), uint(2), "greaterThanIntUintPassTest passes");
     }
-
-    function greaterThanIntUintFailTest() public {
```
Comment (yann300) [inline libs/remix-tests/tests/examples_0/assert_greaterThan_test.sol:33]: this doesn't compile?

File: libs/remix-debug/test/decoder/contracts/miscContracts.ts (hunk 2 above)
Comment (Aniket-Engg) [reply libs/remix-debug/test/decoder/contracts/miscContracts.ts:30]: from https://docs.soliditylang.org/en/v0.8.26/080-breaking-changes.html#new-restrictions

Screenshot 2024-08-21 at 2 42 48 PM

File: libs/remix-tests/tests/examples_0/assert_greaterThan_test.sol (hunk 5 above)
Comment (Aniket-Engg) [reply libs/remix-tests/tests/examples_0/assert_greaterThan_test.sol:33]: In the latest version, `uint` doesn't accept negative values

File: libs/remix-debug/test/decoder/decodeInfo.ts (hunk 3 above)
Comment (Aniket-Engg) [reply libs/remix-debug/test/decoder/decodeInfo.ts:50]: as enum elements in `contractEnum` are reduced

File: libs/remix-debug/test/decoder/localsTests/structArray.ts (hunk 4 above)
Comment (Aniket-Engg) [reply libs/remix-debug/test/decoder/localsTests/structArray.ts]: this is fixed

**Review by yann300 — APPROVED** [review]

**Pull request merged**